src/core/Table/hooks/useColumnVisibility.tsx

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { ColumnInstance, HeaderProps, Hooks } from 'react-table';
+import { Checkbox } from '../..';
+import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
+import { DropdownMenu } from '../../DropdownMenu';
+import { IconButton } from '../../Buttons/IconButton';
+import { MenuItem } from '../../Menu';
+
+export const COLUMN_MANAGER_ID = 'iui-table-column-manager';
+
+export const useColumnVisibility = <T extends Record<string, unknown>>(
+  hasColumnManager: boolean,
+) => (hooks: Hooks<T>) => {
+  if (!hasColumnManager) {
+    return;
+  }
+
+  hooks.allColumns.push((columns: ColumnInstance<T>[]) => [
+    ...columns,
+    {
+      id: COLUMN_MANAGER_ID,
+      minWidth: 48,
+      width: 48,
+      maxWidth: 48,
+      disableReordering: true,
+      columnClassName: 'iui-slot',
+      cellClassName: 'iui-slot',
+      Header: ({ allColumns }: HeaderProps<T>) => {
+        const [isOpen, setIsOpen] = React.useState(false);
+        const headerCheckBoxes = () =>
+          allColumns
+            .filter(({ id }) => !id.includes('iui-table'))
+            .map((column) => {
+              const { onChange, checked } = column.getToggleHiddenProps();
+              return (
+                <MenuItem
+                  key={column.id}
+                  {...column.getToggleHiddenProps()}
+                  icon={
+                    <Checkbox
+                      id={`iui-column-${column.id}`}
+                      checked={checked}
+                      disabled={column.disableToggleVisibility}
+                    />
+                  }
+                  onClick={() => onChange({ target: { checked: !checked } })}
+                  disabled={column.disableToggleVisibility}
+                >
+                  {column.Header}
```

Comment:
hm.. I thought for semantics we wanted to have here `<label>` but I see clicking works fine without it now too.. So maybe no that important (accessibility question)

Reply:
`DropdownMenu` not using labels is not really connected... `label` semantically should be used with any input element. DropdownMenu` just has a prop called label :D
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

So the only concern of mine is a11y and screen readers.

Reply:
As we wont probably use Checkbox anymore, this is not a concern. Resolving